server/driver_tidb.go

```diff
@@ -122,6 +123,7 @@ func (ts *TiDBStatement) Close() error {
 		return errors.Trace(err)
 	}
 	delete(ts.ctx.stmts, int(ts.id))
+	delete(ts.ctx.results, int(ts.id))
```

Comment:
And we also need to Close all the results in connection close to avoid goroutine leak.

Reply:
Updates have been committed.

Closing ResultSet to release resources when Statement or connection is closed.